packages/react-native/Libraries/NativeAnimation/Drivers/RCTAnimationDriver.h

```diff
@@ -9,8 +9,20 @@
 #import <Foundation/Foundation.h>
 
 #import <React/RCTBridgeModule.h>
-
-static CGFloat RCTSingleFrameInterval = (CGFloat)(1.0 / 60.0);
+#import <React/RCTUtils.h>
+
+static CGFloat RCTSingleFrameInterval(void)
+{
+  static CGFloat maximumFramesPerSecond;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    RCTUnsafeExecuteOnMainQueueSync(^{
+      maximumFramesPerSecond = [UIScreen mainScreen].maximumFramesPerSecond;
+    });
```

Comment:
this might deadlock... Do we have a list of devices that supports 120 vs 60?
I kind of think that we don't have to support 60 FPS devices anymore...

Reply:
Hmm I think we still have to support 60fps, only the Pro iPhones come with Pro Motion (120 Hz refresh rate). For example iPhone 16 is still at 60hz. 

Why would it deadlock? We are doing this kind of dispatch_once() that accesses UIScreen quite a lot in the code. 

Looks like JS refers to 60fps here: https://github.com/facebook/react-native/blob/1a6d466f18f0f2163d04008fd3c399cbd4108eb2/packages/react-native/Libraries/Animated/animations/TimingAnimation.js#L89

We would probably need to expose this `maximumFramesPerSecond` to JS